flexmeasures/ui/templates/includes/graphs.html

```diff
@@ -0,0 +1,708 @@
+                    const data = await response.json();
+                    if (data.message.path_and_files === undefined) {
+                        showToast(data.message, "error");
+                    }
+                    else {
+                        for (const [key, value] of Object.entries(data.message.path_and_files)) {
+                            if (typeof value === 'string') {
+                                showToast(value, "error");
+                            } else if (Array.isArray(value)) {
+                                value.forEach(element => {
+                                    showToast(element, "error");
+                                });
+                            } else if (typeof value === 'object' && value !== null) {
+                                Object.values(value).forEach(element => {
+                                    if (Array.isArray(element)) {
+                                        element.forEach(item => showToast(item, "error"));
+                                    } else {
+                                        showToast(element, "error");
+                                    }
+                                });
+                            } else {
+                                // fallback for unexpected types
+                                showToast(String(value), "error");
+                            }
+                        }
+                    }
+                    uploadSpinner.classList.add('d-none');
+                    upload.classList.remove('d-none');
+                }
+            })
+            .catch(function (error) {
+                // Network or unexpected error
+                console.error("Upload error:", error);
+                showToast("An error occurred: " + error.message, "error");
+                uploadSpinner.classList.add('d-none');
+                upload.classList.remove('d-none');
+            });
+    });
+    {% endif %}
+
+
+
+    // Toggle Stats Checkbox Listener
+    {% if active_page == "sensors" %}
+    // Call loadSensorStats if the toggleStatsCheckbox checkbox is checked or unchecked
+    document.getElementById('toggleStatsCheckbox').addEventListener('change', function () {
+        if (this.checked) {
+            loadSensorStats({{ sensor.id }}, storeStartDate.toISOString(), storeEndDate.toISOString());
+                }
+                else {
+        loadSensorStats({{ sensor.id }});
+                }
+            });
+    {% endif %}
+
+    // Keypress Listener
+    window.addEventListener('keypress', function (e) {
+        // Do nothing if typing in an input field or textarea
+        if (e.target.tagName.toLowerCase() === 'input' || e.target.tagName.toLowerCase() === 'textarea') {
+            return;
+        }
+        // Start/pause/resume replay with 'p'
+        if (e.key === 'p') {
+            toggleReplay();
+        } else if (e.key === 's') {
+            stopReplay();
+        }
+    }, false);
```

Comment:
@joshuaunity note for your new PR: this is where the 'n' key stroke should be handled.